extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/ServletProducer.java

```diff
@@ -22,4 +24,10 @@ HttpServletRequest request() {
     HttpServletResponse response() {
         return (HttpServletResponse) ServletRequestContext.requireCurrent().getServletResponse();
     }
+
+    @Produces
+    @SessionScoped
```

Comment:
I think that the scope of this producer method should be `@RequestScoped`. The spec does not define the scope for the built-in bean but if you use `@SessionScoped` then you need to use some kind of proxy because otherwise you would store the `HttpSession` bean instance as an `HttpSession` attribute (which is not a good idea ;-).

CC @manovotn

Reply:
Thank you for the review! I've updated with your suggestion.